src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb

```diff
@@ -2163,25 +2179,16 @@ FullWidthRepeat2:
                ByRef Here As Integer
            ) As Boolean
             Debug.Assert(Here >= 0)
+            Dim ch As Char
 
-            If Not CanGetCharAtOffset(Here) Then
-                Return False
-            End If
-
-            Dim ch = PeekAheadChar(Here)
-            If Not IsDecimalDigit(ch) Then
-                Return False
-            End If
+            If Not TryPeekAheadChar(Here,ch) Then Return False
+            If Not IsDecimalDigit(ch) Then Return False
 
             Dim IntegralValue As Integer = IntegralLiteralCharacterValue(ch)
             Here += 1
 
-            While CanGetCharAtOffset(Here)
-                ch = PeekAheadChar(Here)
-
-                If Not IsDecimalDigit(ch) Then
-                    Exit While
-                End If
+            While TryPeekAheadChar(Here,ch)
+                If Not IsDecimalDigit(ch) Then Exit While
```

Comment:
Inline into the condition of the while - While TryPeekAheadChar AndAlso IsDecimalDigit...